pkg/query/querier.go

```diff
@@ -226,28 +226,41 @@ func (q *querier) Select(_ bool, hints *storage.SelectHints, ms ...*labels.Match
 	}
 
 	// The merged series set assembles all potentially-overlapping time ranges
-	// of the same series into a single one. The series are ordered so that equal series
-	// from different replicas are sequential. We can now deduplicate those.
-	return newDedupSeriesSet(set, q.replicaLabels, len(aggrs) == 1 && aggrs[0] == storepb.Aggr_COUNTER), warns, nil
+	// of the same series into a single one.
+	return newDedupSeriesSet(set, len(aggrs) == 1 && aggrs[0] == storepb.Aggr_COUNTER), warns, nil
 }
 
-// sortDedupLabels re-sorts the set so that the same series with different replica
-// labels are coming right after each other.
-func sortDedupLabels(set []storepb.Series, replicaLabels map[string]struct{}) {
-	for _, s := range set {
+// trimReplicaLabels removed replica labels from all series and re-sorts the set so that the same series are coming right after each other.
+func trimReplicaLabels(set []storepb.Series, replicaLabels map[string]struct{}) {
+	for si := range set {
+		lset := set[si].Labels
```

Comment:
Can we name the `lset` to `labelSet`? Sounds more intuitive

Reply:
I would stick to `lset` I know it's confusing but we are using this name everywhere in Thanos, Prometheus and Cortex. Plus we use `lset` ONLY for label sets (: So with this after learning that `lset` might make sense. What do you think? (:

Reply:
Sure, makes sense to follow the convention 😁